go/vt/vtgate/planbuilder/filter.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package planbuilder
+
+import (
+	"vitess.io/vitess/go/mysql/collations"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/engine"
+	"vitess.io/vitess/go/vt/vtgate/evalengine"
+	"vitess.io/vitess/go/vt/vtgate/semantics"
+)
+
+type (
+	// filter is the logicalPlan for engine.Filter.
+	filter struct {
+		logicalPlanCommon
+		efilter *engine.Filter
+	}
+
+	simpleConverterLookup struct {
+		semTable *semantics.SemTable
+		plan     logicalPlan
+	}
+)
+
+var _ logicalPlan = (*filter)(nil)
+var _ evalengine.ConverterLookup = (*simpleConverterLookup)(nil)
+
+func (s *simpleConverterLookup) ColumnLookup(col *sqlparser.ColName) (int, error) {
```

Comment:
should this be public? if so, it needs a comment

Reply:
The interface has this method in public, I will add descriptive comments to the interface and its implementations